### ISSUE TITLE
Exist 3.2 won't quit on macOS

### ISSUE THREAD

**Issue by duncdrum** [eXist-db/exist#1431]
### What is the problem
After upgrading from 3.1.1 to 3.2 using the dmg installer eXist won't quit after first run.  In addition to the process described [here](https://github.com/eXist-db/exist/issues/1248) which is still happening, now in addition the exist-db process and menubar item do not respond to either "stop server" or "quit and stop server" commands. 

Even after stopping the server remains running, quiting doesn't occur. The only way to properly kill the process is via reboot or force kill in activity monitor. 

After reboot stop quitting sometimes work, sometimes it doesn't. 

### What did you expect
the server to stop and quit when I click stop  and quit.

### Describe how to reproduce or add a test
Previously there were multiple exist processes that stuck around (one for menubar and one for dock). We have limited that down to one,  for the menubar. This is after stop and quit:
![screenshot 2017-05-03 11 48 21](https://cloud.githubusercontent.com/assets/6205362/25655773/6ee2b5da-2ff6-11e7-9d76-bacdbe6edbe0.png)

here is the shutdown log from a failed attempt:
```2017-05-03 14:29:27,237 [AWT-EventQueue-0] INFO  (BrokerPool.java [shutdown]:1518) - Database is shutting down ... 
2017-05-03 14:29:27,741 [AWT-EventQueue-0] INFO  (Journal.java [shutdown]:507) - Transaction journal cleanly shutting down with checkpoint... 
2017-05-03 14:29:27,742 [AWT-EventQueue-0] INFO  (FileLock.java [release]:188) - Deleting lock file: /Users/halalpha/Library/Application Support/org.exist/journal.lck 
2017-05-03 14:29:27,743 [AWT-EventQueue-0] INFO  (FileLock.java [release]:188) - Deleting lock file: /Users/halalpha/Library/Application Support/org.exist/dbx_dir.lck 
2017-05-03 14:29:27,744 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: Scheduler-1005066295 
2017-05-03 14:29:27,744 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: AsyncNCSARequestLog@5275b552 
2017-05-03 14:29:27,744 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: Signal Dispatcher 
2017-05-03 14:29:27,745 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: Java2D Disposer 
2017-05-03 14:29:27,745 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: AWT-Shutdown 
2017-05-03 14:29:27,745 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: Thread-134 
2017-05-03 14:29:27,745 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: Reference Handler 
2017-05-03 14:29:27,745 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: org.eclipse.jetty.util.RolloverFileOutputStream 
2017-05-03 14:29:27,746 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: DestroyJavaVM 
2017-05-03 14:29:27,746 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: TimerQueue 
2017-05-03 14:29:27,746 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: Finalizer 
2017-05-03 14:29:27,746 [AWT-EventQueue-0] INFO  (BrokerPool.java [shutdown]:1646) - shutdown complete ! 
2017-05-03 14:29:27,746 [AWT-EventQueue-0] INFO  (JettyStart.java [shutdown]:554) - Database shutdown: stopping server in 1sec ... 
2017-05-03 14:29:28,752 [jetty shutdown schedule] INFO  (JettyStart.java [lifeCycleStopping]:638) - Jetty server stopping... 
```
=> force quit
here a successful shutdown (same system few minutes later)

```
2017-05-03 14:43:58,952 [AWT-EventQueue-0] INFO  (BrokerPool.java [shutdown]:1518) - Database is shutting down ... 
2017-05-03 14:43:59,382 [AWT-EventQueue-0] INFO  (Journal.java [shutdown]:507) - Transaction journal cleanly shutting down with checkpoint... 
2017-05-03 14:43:59,383 [AWT-EventQueue-0] INFO  (FileLock.java [release]:188) - Deleting lock file: /Users/halalpha/Library/Application Support/org.exist/journal.lck 
2017-05-03 14:43:59,384 [AWT-EventQueue-0] INFO  (FileLock.java [release]:188) - Deleting lock file: /Users/halalpha/Library/Application Support/org.exist/dbx_dir.lck 
2017-05-03 14:43:59,385 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: AWT-Shutdown 
2017-05-03 14:43:59,386 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: Reference Handler 
2017-05-03 14:43:59,386 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: Thread-26 
2017-05-03 14:43:59,386 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: TimerQueue 
2017-05-03 14:43:59,386 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: Scheduler-448147266 
2017-05-03 14:43:59,386 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: Java2D Disposer 
2017-05-03 14:43:59,386 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: org.eclipse.jetty.util.RolloverFileOutputStream 
2017-05-03 14:43:59,387 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: DestroyJavaVM 
2017-05-03 14:43:59,387 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: Finalizer 
2017-05-03 14:43:59,387 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: AsyncNCSARequestLog@72373532 
2017-05-03 14:43:59,387 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: Signal Dispatcher 
2017-05-03 14:43:59,388 [AWT-EventQueue-0] WARN  (BrokerPool.java [clearThreadLocals]:1691) - Could not clear ThreadLocals for thread: qtp2070525344-55 
2017-05-03 14:43:59,388 [AWT-EventQueue-0] INFO  (BrokerPool.java [shutdown]:1646) - shutdown complete ! 
2017-05-03 14:43:59,388 [AWT-EventQueue-0] INFO  (JettyStart.java [shutdown]:554) - Database shutdown: stopping server in 1sec ... 
2017-05-03 14:44:00,392 [jetty shutdown schedule] INFO  (JettyStart.java [lifeCycleStopping]:638) - Jetty server stopping... 
2017-05-03 14:44:00,398 [jetty shutdown schedule] INFO  (AbstractConnector.java [doStop]:310) - Stopped ServerConnector@6a1febd7{HTTP/1.1,[http/1.1]}{0.0.0.0:8080} 
2017-05-03 14:44:00,399 [jetty shutdown schedule] INFO  (AbstractConnector.java [doStop]:310) - Stopped ServerConnector@518f96de{SSL,[ssl, http/1.1]}{0.0.0.0:8443} 
2017-05-03 14:44:00,401 [jetty shutdown schedule] INFO  (ContextHandler.java [doStop]:904) - Stopped o.e.j.w.WebAppContext@5d08f79{/,null,UNAVAILABLE}{/portal} 
2017-05-03 14:44:00,404 [jetty shutdown schedule] INFO  (ContextHandler.java [doStop]:904) - Stopped eXist-db Open Source Native XML Database 
2017-05-03 14:44:00,405 [jetty shutdown schedule] INFO  (JettyStart.java [lifeCycleStopped]:645) - Jetty server stopped 
2017-05-03 14:44:00,405 [BrokerPools-ShutdownHook] INFO  (BrokerPools.java [run]:67) - Executing shutdown thread 
```
### Context information
Please always add the following information
- eXist-db 3.2 
- was Java8u121, now on Java8u131
- MacOs 10.12.4
- 64 bit

**Comment by adamretter** [comment]
So I have found that selecting "Quit (and stop server)" from the eXist menu bar option does seem to stop eXist but certainly does not end the Java processes. I suspect this was also present before 3.2.0 as nothing has changed in this area since before 3.1.1 was released.

After clicking "Quit (and stop server)" I see the following processes still running:
```
501 88122     1   0 11:58am ??         0:10.05 /Applications/eXist-db.app/Contents/MacOS/JavaAppLauncher -psn_0_7440152

501 88148 88122   0 11:58am ??         1:00.19 /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/bin/java -Xms256m -Xmx2048m -Dfile.encoding=UTF-8 -Xdock:name="eXist-db" -Xdock:icon="icon.png" -Dapple.laf.useScreenMenuBar="true" -Dapple.awt.UIElement=true -Dexist.home="/Applications/eXist-db.app/Contents/Resources/eXist-db" -Djava.endorsed.dirs="/Applications/eXist-db.app/Contents/Resources/eXist-db/lib/endorsed" -jar start.jar org.exist.launcher.Launcher
```

The jstack for `88122` reports:
```
Full thread dump Java HotSpot(TM) 64-Bit Server VM (25.121-b13 mixed mode):

"RMI Scheduler(0)" #26 daemon prio=9 os_prio=31 tid=0x00007f893f473000 nid=0x190b waiting on condition [0x0000700006e48000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x000000076aed3918> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1081)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:809)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1067)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1127)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

"RMI TCP Accept-0" #24 daemon prio=9 os_prio=31 tid=0x00007f893dd56800 nid=0xa60f runnable [0x0000700006ab9000]
   java.lang.Thread.State: RUNNABLE
	at java.net.PlainSocketImpl.socketAccept(Native Method)
	at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:409)
	at java.net.ServerSocket.implAccept(ServerSocket.java:545)
	at java.net.ServerSocket.accept(ServerSocket.java:513)
	at sun.management.jmxremote.LocalRMIServerSocketFactory$1.accept(LocalRMIServerSocketFactory.java:52)
	at sun.rmi.transport.tcp.TCPTransport$AcceptLoop.executeAcceptLoop(TCPTransport.java:400)
	at sun.rmi.transport.tcp.TCPTransport$AcceptLoop.run(TCPTransport.java:372)
	at java.lang.Thread.run(Thread.java:745)

"Attach Listener" #23 daemon prio=9 os_prio=31 tid=0x00007f893c877000 nid=0xf957 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"DestroyJavaVM" #22 prio=5 os_prio=31 tid=0x00007f893e2a3000 nid=0x2203 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"yajsw.posix.process.terminate-0" #21 daemon prio=5 os_prio=31 tid=0x00007f893d817800 nid=0xe34b runnable [0x0000700006933000]
   java.lang.Thread.State: RUNNABLE
	at com.sun.jna.Native.invokeInt(Native Method)
	at com.sun.jna.Function.invoke(Function.java:383)
	at com.sun.jna.Function.invoke(Function.java:315)
	at com.sun.jna.Library$Handler.invoke(Library.java:212)
	at com.sun.proxy.$Proxy19.waitpid(Unknown Source)
	at org.rzo.yajsw.os.posix.bsd.macosx.MacOsXProcess$1.run(MacOsXProcess.java:311)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

"TimerQueue" #20 daemon prio=5 os_prio=31 tid=0x00007f893d0c0800 nid=0x11813 waiting on condition [0x0000700006d45000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000006c0d592b8> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
	at java.util.concurrent.DelayQueue.take(DelayQueue.java:211)
	at javax.swing.TimerQueue.run(TimerQueue.java:174)
	at java.lang.Thread.run(Thread.java:745)

"AWT-EventQueue-0" #17 prio=6 os_prio=31 tid=0x00007f893dc91000 nid=0xe0f7 waiting on condition [0x0000700006830000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000006c024bb30> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
	at java.awt.EventQueue.getNextEvent(EventQueue.java:554)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:170)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

"Java2D Disposer" #16 daemon prio=10 os_prio=31 tid=0x00007f893f49f000 nid=0xd60f in Object.wait() [0x000070000672d000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:143)
	- locked <0x00000006c0231258> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:164)
	at sun.java2d.Disposer.run(Disposer.java:148)
	at java.lang.Thread.run(Thread.java:745)

"Java2D Queue Flusher" #15 daemon prio=10 os_prio=31 tid=0x00007f893d566000 nid=0xc207 in Object.wait() [0x000070000662a000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at sun.java2d.opengl.OGLRenderQueue$QueueFlusher.run(OGLRenderQueue.java:203)
	- locked <0x00000006c023c280> (a sun.java2d.opengl.OGLRenderQueue$QueueFlusher)

"AWT-Shutdown" #14 prio=5 os_prio=31 tid=0x00007f893dc8a000 nid=0x7c1b in Object.wait() [0x0000700006421000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.lang.Object.wait(Object.java:502)
	at sun.awt.AWTAutoShutdown.run(AWTAutoShutdown.java:295)
	- locked <0x00000006c02314f8> (a java.lang.Object)
	at java.lang.Thread.run(Thread.java:745)

"AppKit Thread" #13 daemon prio=5 os_prio=31 tid=0x00007f893bc77000 nid=0x307 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Service Thread" #9 daemon prio=9 os_prio=31 tid=0x00007f893f045000 nid=0x5403 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C1 CompilerThread3" #8 daemon prio=9 os_prio=31 tid=0x00007f893b81e000 nid=0x5203 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread2" #7 daemon prio=9 os_prio=31 tid=0x00007f893b826000 nid=0x5003 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread1" #6 daemon prio=9 os_prio=31 tid=0x00007f893d80a000 nid=0x4e03 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread0" #5 daemon prio=9 os_prio=31 tid=0x00007f893f029800 nid=0x4c03 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Signal Dispatcher" #4 daemon prio=9 os_prio=31 tid=0x00007f893f029000 nid=0x4a07 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Finalizer" #3 daemon prio=8 os_prio=31 tid=0x00007f893c809800 nid=0x3c03 in Object.wait() [0x0000700005c09000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:143)
	- locked <0x00000006c024c638> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:164)
	at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:209)

"Reference Handler" #2 daemon prio=10 os_prio=31 tid=0x00007f893f80a800 nid=0x3a03 in Object.wait() [0x0000700005b06000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.lang.Object.wait(Object.java:502)
	at java.lang.ref.Reference.tryHandlePending(Reference.java:191)
	- locked <0x00000006c024c7e0> (a java.lang.ref.Reference$Lock)
	at java.lang.ref.Reference$ReferenceHandler.run(Reference.java:153)

"VM Thread" os_prio=31 tid=0x00007f893b812000 nid=0x3803 runnable 

"GC task thread#0 (ParallelGC)" os_prio=31 tid=0x00007f893d001000 nid=0x2803 runnable 

"GC task thread#1 (ParallelGC)" os_prio=31 tid=0x00007f893d002000 nid=0x2a03 runnable 

"GC task thread#2 (ParallelGC)" os_prio=31 tid=0x00007f893d002800 nid=0x2c03 runnable 

"GC task thread#3 (ParallelGC)" os_prio=31 tid=0x00007f893d003000 nid=0x2e03 runnable 

"GC task thread#4 (ParallelGC)" os_prio=31 tid=0x00007f893d003800 nid=0x3003 runnable 

"GC task thread#5 (ParallelGC)" os_prio=31 tid=0x00007f893d004800 nid=0x3203 runnable 

"GC task thread#6 (ParallelGC)" os_prio=31 tid=0x00007f893d005000 nid=0x3403 runnable 

"GC task thread#7 (ParallelGC)" os_prio=31 tid=0x00007f893b80d000 nid=0x3603 runnable 

"VM Periodic Task Thread" os_prio=31 tid=0x00007f893d801800 nid=0x5603 waiting on condition 

JNI global references: 902
```

Whilst the jstack for 88148 reports:
```
Full thread dump Java HotSpot(TM) 64-Bit Server VM (25.121-b13 mixed mode):

"Attach Listener" #76 daemon prio=9 os_prio=31 tid=0x00007ff24ec8b800 nid=0xc3b3 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"AWT-EventQueue-0" #73 prio=6 os_prio=31 tid=0x00007ff25b5a3000 nid=0x1485b in Object.wait() [0x000070000a370000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.lang.Object.wait(Object.java:502)
	at org.exist.jetty.JettyStart.shutdown(JettyStart.java:533)
	- locked <0x0000000740c94b48> (a org.exist.jetty.JettyStart)
	at org.exist.launcher.Launcher.lambda$createMenu$5(Launcher.java:274)
	at org.exist.launcher.Launcher$$Lambda$37/1469846054.actionPerformed(Unknown Source)
	at java.awt.MenuItem.processActionEvent(MenuItem.java:669)
	at java.awt.MenuItem.processEvent(MenuItem.java:628)
	at java.awt.MenuComponent.dispatchEventImpl(MenuComponent.java:355)
	at java.awt.MenuComponent.dispatchEvent(MenuComponent.java:343)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:761)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:90)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

"pool-2-thread-1" #71 prio=5 os_prio=31 tid=0x00007ff2529b7000 nid=0x15107 waiting on condition [0x000070000c68c000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x0000000797125400> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1093)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:809)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1067)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1127)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

"qtp681742969-62" #62 prio=5 os_prio=31 tid=0x00007ff25cf56000 nid=0x14743 runnable [0x000070000c17d000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.KQueueArrayWrapper.kevent0(Native Method)
	at sun.nio.ch.KQueueArrayWrapper.poll(KQueueArrayWrapper.java:198)
	at sun.nio.ch.KQueueSelectorImpl.doSelect(KQueueSelectorImpl.java:117)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
	- locked <0x00000007aed59e88> (a sun.nio.ch.Util$3)
	- locked <0x00000007aed5dfb0> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000007aed24010> (a sun.nio.ch.KQueueSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:97)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:101)
	at org.eclipse.jetty.io.ManagedSelector$SelectorProducer.select(ManagedSelector.java:243)
	at org.eclipse.jetty.io.ManagedSelector$SelectorProducer.produce(ManagedSelector.java:191)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:249)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:136)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589)
	at java.lang.Thread.run(Thread.java:745)

"qtp681742969-61" #61 prio=5 os_prio=31 tid=0x00007ff2529a0000 nid=0x1391f runnable [0x000070000c07a000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.KQueueArrayWrapper.kevent0(Native Method)
	at sun.nio.ch.KQueueArrayWrapper.poll(KQueueArrayWrapper.java:198)
	at sun.nio.ch.KQueueSelectorImpl.doSelect(KQueueSelectorImpl.java:117)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
	- locked <0x00000007aedd1ba8> (a sun.nio.ch.Util$3)
	- locked <0x00000007aedd1b98> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000007aedd1998> (a sun.nio.ch.KQueueSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:97)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:101)
	at org.eclipse.jetty.io.ManagedSelector$SelectorProducer.select(ManagedSelector.java:243)
	at org.eclipse.jetty.io.ManagedSelector$SelectorProducer.produce(ManagedSelector.java:191)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:249)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:136)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589)
	at java.lang.Thread.run(Thread.java:745)

"qtp681742969-60" #60 prio=5 os_prio=31 tid=0x00007ff25b406000 nid=0x1351f runnable [0x000070000bf77000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.KQueueArrayWrapper.kevent0(Native Method)
	at sun.nio.ch.KQueueArrayWrapper.poll(KQueueArrayWrapper.java:198)
	at sun.nio.ch.KQueueSelectorImpl.doSelect(KQueueSelectorImpl.java:117)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
	- locked <0x00000007aedcbc00> (a sun.nio.ch.Util$3)
	- locked <0x00000007aedcbbf0> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000007aedcb9f0> (a sun.nio.ch.KQueueSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:97)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:101)
	at org.eclipse.jetty.io.ManagedSelector$SelectorProducer.select(ManagedSelector.java:243)
	at org.eclipse.jetty.io.ManagedSelector$SelectorProducer.produce(ManagedSelector.java:191)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:249)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:136)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589)
	at java.lang.Thread.run(Thread.java:745)

"Scanner-0" #49 daemon prio=6 os_prio=31 tid=0x00007ff250b4d800 nid=0x6f0b in Object.wait() [0x000070000be74000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.util.TimerThread.mainLoop(Timer.java:552)
	- locked <0x00000007aeeb2298> (a java.util.TaskQueue)
	at java.util.TimerThread.run(Timer.java:505)

"Scheduler-1066642165" #48 prio=6 os_prio=31 tid=0x00007ff25b060800 nid=0x13303 waiting on condition [0x000070000bd71000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x0000000741cdbdc0> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1093)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:809)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1067)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1127)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

"AsyncNCSARequestLog@30569e72" #47 prio=6 os_prio=31 tid=0x00007ff25b2cf800 nid=0x13103 waiting on condition [0x000070000bc6e000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x0000000741cdc350> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
	at org.eclipse.jetty.util.BlockingArrayQueue.poll(BlockingArrayQueue.java:392)
	at org.eclipse.jetty.server.AsyncNCSARequestLog$WriterThread.run(AsyncNCSARequestLog.java:78)

"org.eclipse.jetty.util.RolloverFileOutputStream" #46 daemon prio=6 os_prio=31 tid=0x00007ff25b2cf000 nid=0x12f03 in Object.wait() [0x000070000bb6b000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.util.TimerThread.mainLoop(Timer.java:552)
	- locked <0x0000000741ce0cb8> (a java.util.TaskQueue)
	at java.util.TimerThread.run(Timer.java:505)

"qtp681742969-44-acceptor-0@43543408-ServerConnector@63b55079{SSL,[ssl, http/1.1]}{0.0.0.0:8443}" #44 prio=5 os_prio=31 tid=0x00007ff24ff54000 nid=0x12b03 runnable [0x000070000b965000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.ServerSocketChannelImpl.accept0(Native Method)
	at sun.nio.ch.ServerSocketChannelImpl.accept(ServerSocketChannelImpl.java:422)
	at sun.nio.ch.ServerSocketChannelImpl.accept(ServerSocketChannelImpl.java:250)
	- locked <0x00000007aef041b0> (a java.lang.Object)
	at org.eclipse.jetty.server.ServerConnector.accept(ServerConnector.java:373)
	at org.eclipse.jetty.server.AbstractConnector$Acceptor.run(AbstractConnector.java:593)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589)
	at java.lang.Thread.run(Thread.java:745)

"qtp681742969-42" #42 prio=5 os_prio=31 tid=0x00007ff259cbc800 nid=0x10f77 runnable [0x000070000b75f000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.KQueueArrayWrapper.kevent0(Native Method)
	at sun.nio.ch.KQueueArrayWrapper.poll(KQueueArrayWrapper.java:198)
	at sun.nio.ch.KQueueSelectorImpl.doSelect(KQueueSelectorImpl.java:117)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
	- locked <0x00000007aed2d778> (a sun.nio.ch.Util$3)
	- locked <0x00000007aed2d768> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000007aed2d648> (a sun.nio.ch.KQueueSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:97)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:101)
	at org.eclipse.jetty.io.ManagedSelector$SelectorProducer.select(ManagedSelector.java:243)
	at org.eclipse.jetty.io.ManagedSelector$SelectorProducer.produce(ManagedSelector.java:191)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:249)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.execute(ExecuteProduceConsume.java:100)
	at org.eclipse.jetty.io.ManagedSelector.run(ManagedSelector.java:147)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589)
	at java.lang.Thread.run(Thread.java:745)

"qtp681742969-41" #41 prio=5 os_prio=31 tid=0x00007ff25b1a1800 nid=0x10907 waiting on condition [0x000070000b65c000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x0000000741b90600> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
	at org.eclipse.jetty.util.BlockingArrayQueue.poll(BlockingArrayQueue.java:392)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.idleJobPoll(QueuedThreadPool.java:563)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.access$800(QueuedThreadPool.java:48)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:626)
	at java.lang.Thread.run(Thread.java:745)

"qtp681742969-40" #40 prio=5 os_prio=31 tid=0x00007ff259cbf800 nid=0x1307 runnable [0x000070000b559000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.KQueueArrayWrapper.kevent0(Native Method)
	at sun.nio.ch.KQueueArrayWrapper.poll(KQueueArrayWrapper.java:198)
	at sun.nio.ch.KQueueSelectorImpl.doSelect(KQueueSelectorImpl.java:117)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
	- locked <0x00000007aedb2ad8> (a sun.nio.ch.Util$3)
	- locked <0x00000007aedb2ac8> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000007aedb2998> (a sun.nio.ch.KQueueSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:97)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:101)
	at org.eclipse.jetty.io.ManagedSelector$SelectorProducer.select(ManagedSelector.java:243)
	at org.eclipse.jetty.io.ManagedSelector$SelectorProducer.produce(ManagedSelector.java:191)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:249)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.execute(ExecuteProduceConsume.java:100)
	at org.eclipse.jetty.io.ManagedSelector.run(ManagedSelector.java:147)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589)
	at java.lang.Thread.run(Thread.java:745)

"qtp681742969-39" #39 prio=5 os_prio=31 tid=0x00007ff24ff52800 nid=0xde07 runnable [0x000070000b456000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.KQueueArrayWrapper.kevent0(Native Method)
	at sun.nio.ch.KQueueArrayWrapper.poll(KQueueArrayWrapper.java:198)
	at sun.nio.ch.KQueueSelectorImpl.doSelect(KQueueSelectorImpl.java:117)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
	- locked <0x00000007aedb2f20> (a sun.nio.ch.Util$3)
	- locked <0x00000007aedb2f10> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000007aedb2de0> (a sun.nio.ch.KQueueSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:97)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:101)
	at org.eclipse.jetty.io.ManagedSelector$SelectorProducer.select(ManagedSelector.java:243)
	at org.eclipse.jetty.io.ManagedSelector$SelectorProducer.produce(ManagedSelector.java:191)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:249)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.execute(ExecuteProduceConsume.java:100)
	at org.eclipse.jetty.io.ManagedSelector.run(ManagedSelector.java:147)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589)
	at java.lang.Thread.run(Thread.java:745)

"qtp681742969-38" #38 prio=5 os_prio=31 tid=0x00007ff24ff52000 nid=0x1107 runnable [0x000070000abbb000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.KQueueArrayWrapper.kevent0(Native Method)
	at sun.nio.ch.KQueueArrayWrapper.poll(KQueueArrayWrapper.java:198)
	at sun.nio.ch.KQueueSelectorImpl.doSelect(KQueueSelectorImpl.java:117)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
	- locked <0x00000007aedd63e8> (a sun.nio.ch.Util$3)
	- locked <0x00000007aedd63d8> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000007aedd61d8> (a sun.nio.ch.KQueueSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:97)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:101)
	at org.eclipse.jetty.io.ManagedSelector$SelectorProducer.select(ManagedSelector.java:243)
	at org.eclipse.jetty.io.ManagedSelector$SelectorProducer.produce(ManagedSelector.java:191)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:249)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:136)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589)
	at java.lang.Thread.run(Thread.java:745)

"qtp681742969-36" #36 prio=5 os_prio=31 tid=0x00007ff24faee000 nid=0x12707 runnable [0x0000700008ca7000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.KQueueArrayWrapper.kevent0(Native Method)
	at sun.nio.ch.KQueueArrayWrapper.poll(KQueueArrayWrapper.java:198)
	at sun.nio.ch.KQueueSelectorImpl.doSelect(KQueueSelectorImpl.java:117)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
	- locked <0x00000007aedb2690> (a sun.nio.ch.Util$3)
	- locked <0x00000007aedb2680> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000007aedb2550> (a sun.nio.ch.KQueueSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:97)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:101)
	at org.eclipse.jetty.io.ManagedSelector$SelectorProducer.select(ManagedSelector.java:243)
	at org.eclipse.jetty.io.ManagedSelector$SelectorProducer.produce(ManagedSelector.java:191)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:249)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.execute(ExecuteProduceConsume.java:100)
	at org.eclipse.jetty.io.ManagedSelector.run(ManagedSelector.java:147)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589)
	at java.lang.Thread.run(Thread.java:745)

"TimerQueue" #23 daemon prio=5 os_prio=31 tid=0x00007ff250205800 nid=0x107f3 waiting on condition [0x000070000a9b5000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x0000000740cde868> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
	at java.util.concurrent.DelayQueue.take(DelayQueue.java:211)
	at javax.swing.TimerQueue.run(TimerQueue.java:174)
	at java.lang.Thread.run(Thread.java:745)

"Java2D Queue Flusher" #22 daemon prio=10 os_prio=31 tid=0x00007ff24ee32000 nid=0xca07 in Object.wait() [0x000070000a6a6000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at sun.java2d.opengl.OGLRenderQueue$QueueFlusher.run(OGLRenderQueue.java:203)
	- locked <0x00000007401b0b60> (a sun.java2d.opengl.OGLRenderQueue$QueueFlusher)

"Java2D Disposer" #21 daemon prio=10 os_prio=31 tid=0x00007ff251c55000 nid=0xc403 in Object.wait() [0x000070000a5a3000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x000000074019b148> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:143)
	- locked <0x000000074019b148> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:164)
	at sun.java2d.Disposer.run(Disposer.java:148)
	at java.lang.Thread.run(Thread.java:745)

"DestroyJavaVM" #13 prio=5 os_prio=31 tid=0x00007ff24f9d5800 nid=0x1c03 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"AWT-Shutdown" #11 prio=5 os_prio=31 tid=0x00007ff250363800 nid=0x7e1f in Object.wait() [0x000070000a1ec000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x00000007401c87d0> (a java.lang.Object)
	at java.lang.Object.wait(Object.java:502)
	at sun.awt.AWTAutoShutdown.run(AWTAutoShutdown.java:295)
	- locked <0x00000007401c87d0> (a java.lang.Object)
	at java.lang.Thread.run(Thread.java:745)

"AppKit Thread" #10 daemon prio=5 os_prio=31 tid=0x00007ff251b02000 nid=0x307 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Service Thread" #9 daemon prio=9 os_prio=31 tid=0x00007ff24e80a000 nid=0x5303 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C1 CompilerThread3" #8 daemon prio=9 os_prio=31 tid=0x00007ff25004e800 nid=0x5103 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread2" #7 daemon prio=9 os_prio=31 tid=0x00007ff250045800 nid=0x4f03 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread1" #6 daemon prio=9 os_prio=31 tid=0x00007ff250042800 nid=0x4d03 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread0" #5 daemon prio=9 os_prio=31 tid=0x00007ff251834800 nid=0x4b03 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Signal Dispatcher" #4 daemon prio=9 os_prio=31 tid=0x00007ff251800000 nid=0x4903 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Finalizer" #3 daemon prio=8 os_prio=31 tid=0x00007ff24f824800 nid=0x3903 in Object.wait() [0x0000700009951000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:143)
	- locked <0x00000007401c9258> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:164)
	at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:209)

"Reference Handler" #2 daemon prio=10 os_prio=31 tid=0x00007ff24f809800 nid=0x3703 in Object.wait() [0x000070000984e000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.lang.Object.wait(Object.java:502)
	at java.lang.ref.Reference.tryHandlePending(Reference.java:191)
	- locked <0x00000007401b0fd8> (a java.lang.ref.Reference$Lock)
	at java.lang.ref.Reference$ReferenceHandler.run(Reference.java:153)

"VM Thread" os_prio=31 tid=0x00007ff25001a000 nid=0x3503 runnable 

"GC task thread#0 (ParallelGC)" os_prio=31 tid=0x00007ff24f804800 nid=0x2503 runnable 

"GC task thread#1 (ParallelGC)" os_prio=31 tid=0x00007ff24f805000 nid=0x2703 runnable 

"GC task thread#2 (ParallelGC)" os_prio=31 tid=0x00007ff24f806000 nid=0x2903 runnable 

"GC task thread#3 (ParallelGC)" os_prio=31 tid=0x00007ff24f806800 nid=0x2b03 runnable 

"GC task thread#4 (ParallelGC)" os_prio=31 tid=0x00007ff24f807000 nid=0x2d03 runnable 

"GC task thread#5 (ParallelGC)" os_prio=31 tid=0x00007ff252000000 nid=0x2f03 runnable 

"GC task thread#6 (ParallelGC)" os_prio=31 tid=0x00007ff24f807800 nid=0x3103 runnable 

"GC task thread#7 (ParallelGC)" os_prio=31 tid=0x00007ff24f808800 nid=0x3303 runnable 

"VM Periodic Task Thread" os_prio=31 tid=0x00007ff250028800 nid=0x5503 waiting on condition 

JNI global references: 1111
```

**Comment by duncdrum** [comment]
@adamretter Glad you concur, however:
> I suspect this was also present before 3.2.0 as nothing has changed in this area since before 3.1.1 was released.

is not what i am seeing on 3.1.1 after selecting quit and stop, the process stops. Also try to connect to the stopped database in 3.2 via oxygen, i could frequently still perform actions connecting to 3.2 via oxygen after minutes after selecting stop and quit, not so with 3.1.1

**Comment by duncdrum** [comment]
This [ticket](https://github.com/eXist-db/dashboard/issues/37) might also be relevant.

**Comment by duncdrum** [comment]
3.3 updating from 3.1.1 to 3.3 via bintray .dmg image on the initial run "Quit (and stop server)" worked (the menu-bar item close and bothy jetty and exist completed their shutdown)! However, the process ``exist-JavaAppLauncher`` silently stuck around. After relogging OS user we are back at the 3.2 behaviour where shutdown won't complete if triggered via menu-bar, and processes have to be force-quit manually.

**Comment by duncdrum** [comment]
tested with exist 3.4, on Java 1.8.0.141 and macOS 10.12.6 with the same result

**Comment by joewiz** [comment]
I forgot that @adamretter and @shabanovd had suggested in https://github.com/eXist-db/exist/issues/1248#issuecomment-285598096 that we try grabbing the stack of the ghost process. That's probably the next best step.

**Comment by joewiz** [comment]
@duncdrum Wait, really, you still get into a state where eXist won't shut down? It's not just the case that the eXist-db dock icon won't automatically quit during first install?

**Comment by duncdrum** [comment]
@joewiz I'll run through a few reinstalls to test this more throughly this week. I had focused on testing for the other mac issue. Since 3.3. the menu-bar icon disappears, (it stuck around in 3.2) but there is still a lingering process that I need to force-quit. On my two installs of 3.4 I had the same behaviour. This is not the java-app-launcher, that remains in the dock, on initial run, which you reported. More soon

**Comment by joewiz** [comment]
Sounds good! I haven't seen this on any of the 3 machines I run, since updating to 3.4.0 (I am running current macOS and Java JDK). I would be happy to test your steps if you have some that trigger it.

**Comment by duncdrum** [comment]
Ok outside of #1248 this seems fixed. No clue why the non java app launcher process stuck around in activity monitor after the menu bar item had disappeared, but I can't reliably reproduce this anymore. Since it was already only a cosmetic problem since 3.3, i ll open a new ticket if it comes back.

**Comment by joewiz** [comment]
Phew! That's good news. Thank you for checking.
